### PR TITLE
test: Cypress - added  tags - @tag.IDE, @tag.JS

### PR DESCRIPTION
**Pull request by sharanya-appsmith**
-  cypress/e2e/Regression/ClientSide/IDE
- cypress/e2e/Regression/ClientSide/JSObject

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?
- [ ] Manual
- [ ] Cypress

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Tests**
	- Enhanced test suite descriptions and added categorization tags for better filtering.
	- Improved test code structure and added explanatory comments for increased clarity and maintainability.

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/JSObject/Bug9585_spec.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { jsEditor } from "../../../../support/Objects/ObjectsCore";
 
-describe("List no functions on empty collection", () => {
+describe("List no functions on empty collection", { tags: ["@tag.JS"] }, () => {
```
Comment (mohanarpit) [inline app/client/cypress/e2e/Regression/ClientSide/JSObject/Bug9585_spec.ts:3]: @sharanya-appsmith Minor: Can we rename this spec with a better name?

Comment (sharanya-appsmith) [reply app/client/cypress/e2e/Regression/ClientSide/JSObject/Bug9585_spec.ts:3]: Done. Renamed to JS_Message_Bug9585_spec.ts

Comment (mohanarpit) [reply app/client/cypress/e2e/Regression/ClientSide/JSObject/Bug9585_spec.ts:3]: @sharanya-appsmith That wasn't my suggestion. As we discussed last week, can we please rename these files and remove the words "Bug9585" in the title? It doesn't add value and doesn't help me understand what this spec is all about. 

It's perfectly ok to comment about the bug inside the spec, but the title/file name should be a quick handy reference to give me a sense about the spec itself.

Comment (sharanya-appsmith) [reply app/client/cypress/e2e/Regression/ClientSide/JSObject/Bug9585_spec.ts:3]: Removed Bug* and renamed to JS_Editor_Assert_NoFunctionAvailableText_spec.ts

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**